### PR TITLE
Repeating Tracemalloc Benchmark for accuracy

### PR DESCRIPTION
**Pull request by trexfeathers**
~~A demonstration of how we can use ASV's built-in repetition logic to improve the accuracy of `tracemalloc` benchmarking.~~

~~**I welcome anyone who wants to finish this off** - I have some other things I need to focus on at the moment.~~

## Edit `2024-07-05`

Recovering from a disastrous afternoon yesterday, I believe I now have this ready to go. I have tested it and it definitely does benefit from the repeat behaviour by detecting when accuracy prevents certainty about a regression. The in-built reporting for byte measurements is also quite nice, including automatic choice between `KB` / `MB` / etcetera.

### CODE REVIEW

**Comment by trexfeathers** [comment]
I will have this reviewable by Thursday '2024-07-04`

File: benchmarks/custom_bms/README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Iris custom benchmarks
+
+Structured and written in accordance with the [ASV guidelines](https://asv.readthedocs.io/projects/asv-runner/en/latest/development/benchmark_plugins.html).
+
+To work, these benchmarks must be installed into the environment where the
+benchmarks are run (i.e. not the environment containing ASV + Nox, but the
+one built to the same specifications as the Tests environment). This is done
+by modifying `sys.path` in 
+[the benchmarks `__init__`](../benchmarks/__init__.py).
```
Comment (trexfeathers) [inline benchmarks/custom_bms/README.md:9]: No longer true, I will need to modify this

**Comment by trexfeathers** [comment]
We're up to 7 issues that this will close 📈

**Review by HGWright — APPROVED** [review]
Thanks @trexfeathers hopefully this removes some of the noise from memory benchmarks going forward!

**Comment by trexfeathers** [comment]
Thanks @HGWright!

**Pull request merged**